### PR TITLE
Nearly one to one clone of `Data.EitherR`

### PR DESCRIPTION
**Pull request by paluh**

### CODE REVIEW

File: src/Data/EitherR.purs [hunk 1]
```diff
@@ -0,0 +1,186 @@
+-- | This module provides `throwEither` and `cachEither` for Either. These
+-- | functions reside here because `throwEither` and `catchEither` correspond
+-- | to `return` and `bind` for flipped `Either` monad: `EitherR`.
+-- |
+-- | `catchEither` differs from `MonadError` (`cacheError`) - `catchEither` is
+-- | more general as it allows you to cahnge the left value's type.
```
Comment (passy) [inline src/Data/EitherR.purs]: s/cahnge/change/ :)

File: test/Main.purs [hunk 2]
```diff
@@ -1,100 +1,21 @@
...
-      let unwrap = runExcept res
-      case unwrap of
-        Right a -> a == "something"
-        _       -> false
-
-  test "(?:)" $ do
-    Assert.assert "from Nothing" $ (Nothing ?: "nothing") == "nothing"
-    Assert.assert "from Just" $ (Just "something" ?: "nothing") == "something"
+  test "Control.Error.Util" Test.Control.Error.Util.suite
+  test "Data.EitherR" Test.Data.EitherR.suite
```
Comment (passy) [inline test/Main.purs:21]: Sweet! Thanks for splitting this up! 👍

**Comment by passy** [comment]
💥 Landed!

Thanks so much, @paluh! I rarely see quality PRs like this. From asking before opening the pull request to adding tests and comprehensive docs. Fantastic work!

**Comment by paluh** [comment]
Thanks!

In case of any bugs I would happily provide regression tests ;-)

Thanks for this port of Gabriel's library - it is really handy!

Till next PR :-)

**Pull request closed without merging**